pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/SingleConnectionBrokerRequestHandler.java

```diff
@@ -159,11 +164,14 @@ protected BrokerResponseNative processBrokerRequest(long requestId, BrokerReques
     }
     int numServersNotResponded = serversNotResponded.size();
     if (numServersNotResponded != 0) {
-      brokerResponse.setPartialResult(true);
       brokerResponse.addToExceptions(new QueryProcessingException(QueryException.SERVER_NOT_RESPONDING_ERROR_CODE,
           String.format("%d servers %s not responded", numServersNotResponded, serversNotResponded)));
       _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.BROKER_RESPONSES_WITH_PARTIAL_SERVERS_RESPONDED, 1);
     }
+    if (totalServersFailedOrNotResponded > 0) {
```

Comment:
why do we need this? we can simply do
```suggestion
    if (numServersQueried - numServersResponded > 0) {
```
if there's any exception inside the response, i thought the broker reducer is going to rectify and add proper partial result or failure inside?

Reply:
chatted with @Jackie-Jiang offline. i think the idea is to set this as "partial response" even when the results are completely correct. we still want to surface a global "flag" indicating that 'something can be improved underneath'.

Reply:
@Jackie-Jiang @walterddr This was added to make sure any exceptions in the returned dataTable (even if the server has responded with a valid set of rows), are also flagged as potentially partial

Reply:
This should not be handled here. We can simply check `if (brokerResponse.getExceptionsSize() > 0)` in the end in `BaseBrokerRequestHandler` and set the flag.